sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -3304,6 +3304,26 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession {
         """.stripMargin).collect()
     }
   }
+
+  test("SPARK-29860: Fix dataType mismatch issue for InSubquery") {
+    withTempView("ta", "tb", "tc", "td") {
+      sql("create temporary view ta as select * from values(cast(1 as Decimal(8, 0))) as ta(id)")
```

Comment:
nit: plz capitalize SQL keywords, e.g., CREATE, SELECT...

Reply:
thanks, will update it in next commit.